test/distributed/tensor/test_utils.py

```diff
@@ -766,6 +774,61 @@ def test_2d_mesh_uneven_strided_shard(self):
             self.assertEqual(dtensor.full_tensor(), tensor)
 
 
+class Test_StridedShard_with_shard_order(LocalDTensorTestBase):
+    @property
+    def world_size(self) -> int:
+        return 32
+
+    @with_comms
+    def test_StridedShard_to_shard_order(self):
+        with LocalTensorMode(ranks=self.world_size):
+            mesh = DeviceMesh("cpu", torch.arange(self.world_size).view(2, 2, 2, 2, 2))
+            shard_iter = generate_shard_orders(mesh, 3)
+            # It takes ~4.8h to complete total 2520 shard order combinations here
+            # using LocalTensor. So we only randomly pick 25 shard orders to test.
+            all_shard_order = list(shard_iter)
+            import random
+
+            shard_order_choices = random.sample(
```

Comment:
i'd rather not have the test be nondeterministic.  perhaps take a snapshot (or seed this) so we are getting some fixed sized problem that is repeatable?